scheduler/scheduler.go

```diff
@@ -128,7 +131,17 @@ type GroupedLIFOFilter interface {
 // It is mandatory to call done() the request was processed. When the
 // request needs to be rejected, an error will be returned.
 func (q *Queue) Wait() (done func(), err error) {
-	return q.queue.Wait()
+	done, err = q.queue.Wait()
+	if q.metrics != nil && err != nil {
+		switch err {
+		case jobqueue.ErrStackFull:
+			q.metrics.IncCounter(q.rejectedRequestsMetricsKey)
+		case jobqueue.ErrTimeout:
+			q.metrics.IncCounter(q.timedOutRequestsMetricsKey)
+		}
+	}
```

Comment:
I think maybe we should name metrics after error types here like
- `lifo.%s.error.full`
- `lifo.%s.error.timeout`
- `lifo.%s.error.other` (missing default case)

Lets ask @aryszka and @szuecs 

As a note for prometheus metrics we could use single metric name (e.g. `lifo_errors`) and put error type (as well as queue name) as a label but I guess we are stuck with existing metrics interface that only supports metrics key for now.

Reply:
Hey @AlexanderYastrebov 
I am fine with the schema for the keys, thanks for the suggestion.
However, I feel like changing how the metrics interface works is out of scope as my only goal is to record these metrics.

Should I wait for more feedback or implement the changes?

Reply:
@wndhydrnt Metrics interface refactoring is out of scope of course, its just a note.

The proposal does not measure other kind of errors besides full queue and timeout. Also my idea was to highlight in the key that these are error metrics compared to regular metrics like
https://github.com/zalando/skipper/blob/2059778e1924aaafb86accfc0b27e10245f13117/scheduler/scheduler.go#L195-L196

> Should I wait for more feedback or implement the changes?

You may do the changes if you agree. We would need another maintainer to have a look anyway.

Reply:
Hello @AlexanderYastrebov 
I finally got around to implementing the changes. Besides your suggestions, the names of the new `*MetricsKey` variables have been changed because they got awfully long.
Please take a look.